lib/hl7/message.ex

```diff
@@ -145,6 +153,35 @@ defmodule HL7.Message do
     msg
   end
 
+  def new(raw_text, %{validate_string: true, accept_latin1: true} = options) do
```

Comment:
Nitpick on the implementation, define a:

```suggestion
    def new(raw_text, options) do
      from_raw(raw_text, Map.get(options, :validate_string, false), Map.get(options, :accept_latin, false))
    end
```

And I think the actual implementation will be clearer as it will do less map mangling?

Reply:
Ah, you may need to pass copy too.

Reply:
@josevalim Thanks! I ended up refactoring out the map mangling to make a `with` statement that could handle the encoding and validation or return an invalid message and it's much easier to follow now i think :)